components/permalink_view/actions.ts

```diff
@@ -81,6 +87,11 @@ export function focusPost(postId: string, returnTo = '', currentUserId: string)
             return;
         }
 
+        if (data.first_inaccessible_post_time) {
+            browserHistory.replace(`/error?type=${ErrorPageTypes.CLOUD_ARCHIVED}&returnTo=${returnTo}`);
```

Comment:
I'm wondering if this browser redirect is the correct way to approach this. I'm thinking of a specific scenario where you might have a link to a post in a thread, but you don't actually want to see that specific post, you want to see the thread in general. But because this post is inaccessible, the whole thread link is broken (as it redirects, even though later posts in the thread may still be accessible).... 

This may be intentional, as it's a driving force to upgrade, but something I wanted to call out.

Reply:
Its intentional, but more for simplicity reasons than a deliberate product desision. Because the root post will always be archived, you have to "fake" a root post in order to display it, and that ends up being a lot more error prone and needing for a lot of places to change in the code. I started down that road but it was getting really messy.